File: Python/263.Ugly Number.py
#判断一个数是否是Ugly Number，定义为因数只包含2,3,5的。
#特殊的，1是Ugly Number。
class Solution(object):
    def isUgly(self, num):
        if num==0:  return False
        if num==1:  return True
        while num%2==0:     num = num//2
        while num%3==0:     num = num//3
        while num%5==0:     num = num//5
        return num==1

s = Solution()
print(s.isUgly(14))

File: C++/Math/263.Ugly Number.cpp
/*
判断一个数是否是Ugly Number，定义为因数只包含2,3,5的。
特殊的，1是Ugly Number。
*/

#include <iostream>
using namespace std;

class Solution {
public:
	bool isUgly(int num) {
        if (num==0) return false;
        while (num%2==0)     num /= 2;
        while (num%3==0)     num /= 3;
        while (num%5==0)     num /= 5;
        return num==1;
    }
};
int main()
{
	Solution s;
	cout<<s.isUgly(12345)<<endl;
}

